### PR TITLE
Now with subdomains, I guess

### PR DESCRIPTION
**Pull request by indirect**

### CODE REVIEW

**Comment by indirect** [comment]
Look, ma! PR-specific preview deployments!

File: app/controllers/feeds_controller.rb [hunk 1]
```diff
@@ -4,10 +4,16 @@ def new
...
   def show
-    @feed = Feed.find_by!(token: params[:id])
+    @feed = Feed.find_by!(token: params[:id] || request.subdomain)
     @feed.touch(:fetched_at) if request.format == :atom
   end
 
+  def favicon
+    return redirect_to helpers.asset_path("favicon.ico") if request.subdomain.blank?
+
+    redirect_to Feed.find_by!(token: request.subdomain).favicon_url, allow_other_host: true
```
Comment (numist) [inline app/controllers/feeds_controller.rb:14]: How confused is 300ing the favicon going to make feed readers? Hopefully not at all, but I can see this needing more attention (proxying the favicon data? storing it in the db?) down the line 😬

File: app/views/feeds/show.html.erb [hunk 2]
```diff
@@ -9,7 +9,7 @@
   <%= tag.div id: dom_id(@feed), class: "max-w-xl mt-5 mx-auto" do %>
     <div class="my-8">
       <h4 class="block text-md font-bold text-gray-700">Email address</h4>
-      <pre class="my-2"><code class="bg-slate-100 border-slate-300 rounded px-4 py-2"><%= @feed.token %>@<%= request.host %></code></pre>
+      <pre class="my-2"><code class="bg-slate-100 border-slate-300 rounded px-4 py-2"><%= @feed.token %>@<%= request.domain %></code></pre>
```
Comment (numist) [inline app/views/feeds/show.html.erb:12]: TIL!

Given the existence of "top level" things like `.co.uk` I kinda wish we could express our intention more clearly by passing `[1..-1]` or something to [`domain`](https://apidock.com/rails/ActionDispatch/Http/URL/domain).

For that matter, how does this interact with the existence of a `staging` subdomain?

File: config/initializers/host.rb [hunk 3]
```diff
@@ -1,8 +1,13 @@
-if ENV.key?("HOST")
-  Rails.application.config.hosts << ENV["HOST"]
+def set_up_host(host)
+  return unless host
```
Comment (numist) [inline config/initializers/host.rb:2]: Should this `raise`? Seems bad.

**Review by numist — COMMENTED** [review]
Couple questions. While we're here should we also strip the og/twitter card stuff out of the template when the page being rendered has a non-FYE favicon?

File: app/controllers/feeds_controller.rb (hunk 1 above)
Comment (indirect) [reply app/controllers/feeds_controller.rb:14]: we can switch to proxying it as a one-line change, but storing it in our DB is probably the way to go in the end.

File: config/initializers/host.rb (hunk 3 above)
Comment (indirect) [reply config/initializers/host.rb:2]: when you run `bin/dev`, and it runs at localhost:3000, that's when HOST isn't set and we need to return. when we deploy, if a request comes in and we didn't add the right domain to the `config.hosts` allow-list, everything will by a 500 and we will definitely notice. we could `raise if Rails.env.production?` though!

File: app/views/feeds/show.html.erb (hunk 2 above)
Comment (indirect) [reply app/views/feeds/show.html.erb:12]: yeah, so the way that works is a global configuration of how many bits to chop off:

Comment (indirect) [reply app/views/feeds/show.html.erb:12]: oops meant to link to it. here's the global config that looks at HOST to decide how many ending TLD bits to chop off: https://github.com/indirect/feedyour.email/pull/42/files#diff-d2c69534ceb90069932d2ec9d4957bc8860e8af5fbdbf04064eb6ed635d76300R6-R7

If HOST is feedyour.email, then `subdomain` will return `foo` for `foo.feedyour.email`. If HOST is staging.feedyour.email, then `subdomain` will return `foo` for `foo.staging.feedyour.email`.

Comment (numist) [reply app/views/feeds/show.html.erb:12]: so… if HOST doesn't include the token does that mean we actually still want `host` here?

**Comment by indirect** [comment]
Hopefully we can get away without subdomains via a combination of JSON Feeds and <link> tags with better icons. 🤞🏻

**Pull request closed without merging**